core/src/main/java/gyro/core/diff/Diff.java

```diff
@@ -214,7 +214,7 @@ private Change newUpdate(Diffable currentDiffable, Diffable pendingDiffable) {
         for (DiffableField field : DiffableType.getInstance(currentDiffable.getClass()).getFields()) {
 
             // Skip nested diffables since they're handled by the diff system.
-            if (field.shouldBeDiffed()) {
+            if (field.shouldBeDiffed() || field.isImmutable()) {
```

Comment:
I would make this a separate `if` check with comment to be consistent with the other checks.